spec.html

```diff
@@ -4681,19 +4681,21 @@ <h1 id="ao-issuperreference">IsSuperReference ( _V_ )</h1>
           1. Set _method_ to ? GetMethod(_obj_, @@iterator).
         1. Let _iterator_ be ? Call(_method_, _obj_).
         1. If Type(_iterator_) is not Object, throw a *TypeError* exception.
-        1. Return _iterator_.
+        1. Let _nextMethod_ be ? GetV(_iterator_, `"next"`).
```

Comment:
I left the check at the call site to avoid changing more than what was needed. I'd be happy to make that change if that's the preferred behavior, however.

Reply:
Fair enough, I'll change.

Reply:
I don't think this should use GetMethod.

"next" isn't an optional method, so treating "null" specially isn't really required, and since the method isn't optional, the only real impact of it would be the error message saying "next" is undefined for an iterator like `{ next: null }`.

Reply:
I think you're right @caitp, we shouldn't use `GetMethod`. I was under the impression that `GetMethod` returned a either a function or an exception. It seems to me that we should either do `If IsCallable(nextMethod) is false, throw a TypeError exception.` after the `GetV` or just wait until the the exception is thrown when trying to call `nextMethod`. I'm slightly more partial to the latter since it's a little less spec text and might be a little nicer for implementations.